library/spdm_requester_lib/libspdm_req_get_csr.c

```diff
@@ -65,17 +65,26 @@ static libspdm_return_t libspdm_try_get_csr(libspdm_context_t *spdm_context,
     }
 
     if (libspdm_get_connection_version(spdm_context) >= SPDM_MESSAGE_VERSION_13) {
+        const uint8_t csr_cert_model = request_attribute &
+                                       SPDM_GET_CSR_REQUEST_ATTRIBUTES_CERT_MODEL_MASK;
+
         /* CSR_CAP for a 1.2 Responder is not checked because it was not defined in SPDM 1.2.0. */
         if (!libspdm_is_capabilities_flag_supported(
                 spdm_context, true, 0,
                 SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CSR_CAP)) {
             return LIBSPDM_STATUS_UNSUPPORTED_CAP;
         }
-        if ((spdm_context->connection_info.multi_key_conn_rsp) == (key_pair_id == 0)) {
-            return LIBSPDM_STATUS_INVALID_PARAMETER;
-        }
-        if (!spdm_context->connection_info.multi_key_conn_rsp) {
-            if ((request_attribute & SPDM_GET_CSR_REQUEST_ATTRIBUTES_CERT_MODEL_MASK) != 0) {
+        if (spdm_context->connection_info.multi_key_conn_rsp) {
+            if (key_pair_id == 0) {
+                return LIBSPDM_STATUS_INVALID_PARAMETER;
+            }
+            if ((csr_cert_model != SPDM_CERTIFICATE_INFO_CERT_MODEL_DEVICE_CERT) &&
```

Comment:
Hi. It is better to use:
req_cert_model >= SPDM_GET_CSR_REQUEST_ATTRIBUTES_MAX_CSR_CERT_MODEL

Reply:
It can't be `0` either. I'll make a helper function in `common_lib` and then this logic can be consolidated.

Reply:
Didn't make a helper function for now.